src/lib/config/merge.ts

```diff
@@ -11,7 +11,7 @@ import { type Configuration } from './types';
 /**
  * Default merge rules for [mergeConfig](#mergeConfig), which are:
  *
- * - {@link exclude}, {@link excludeGlob} and {@link keepUnused} arrays will be concatenated instead of replaced;
+ * - {@link exclude}, {@link excludeGlob}, {@link overrides} and {@link keepUnused} arrays will be concatenated instead of replaced;
```

Comment:
Why do you concat `overrides` instead of replace?
How the order is defined when concatting since order matters in the array?